### PR TITLE
Add Use Cases and NFRs to DeveloperGuide.md

### PR DESCRIPTION
**Pull request by ling-ui**
### Changes

- Added all the use cases for features set to be implemented in v1.2 to DeveloperGuide.md 
- Added some NFRs based on our current features

While reviewing, check for:

- Grammar errors and punctuation errors.
- Whether the numbering and labels are correct
- Any other formatting errors.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -283,16 +283,37 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `TruthTable` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: Add a member to a team**
```
Comment (Devanshshah1309) [inline docs/DeveloperGuide.md:288]: Do we have to number the use cases (e.g. use case 1, use case 2, etc.)

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -304,19 +325,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-    * 3a1. AddressBook shows an error message.
+    * 3a1. TruthTable shows an error message.
 
       Use case resumes at step 2.
 
-*{More to be added}*
+**Use case: List all members of a team**
 
-### Non-Functional Requirements
+Preconditions: The current working team is set to the team that the member should be deleted from.
```
Comment (Devanshshah1309) [inline docs/DeveloperGuide.md:334]:
```suggestion
Precondition: The current working team is set to the team that the member should be deleted from.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -304,19 +325,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 1c. The task deadline is badly formatted.
+
+    * 1c1. TruthTable shows an error message.
+
+      Use case resumes at step 1.
+
+**Use case: Delete a task from a team**
+
+Preconditions: The current working team is set to the team that the task should be deleted from.
```
Comment (Devanshshah1309) [inline docs/DeveloperGuide.md:382]:
```suggestion
Precondition: The current working team is set to the team that the task should be deleted from.
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -304,19 +325,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 3a. The given index is invalid.
+
+    * 3a1. TruthTable shows an error message.
+
+      Use case resumes at step 2.
+
+**Use case: List all tasks of a team**
+
+Preconditions: The current working team is set to the team that the member should be deleted from.
```
Comment (Devanshshah1309) [inline docs/DeveloperGuide.md:407]:
```suggestion
Precondition: The current working team is set to the team that the member should be deleted from.
```

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -304,19 +325,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+**Use case: Add deadline to existing task**
+
+Preconditions: The current working team is set to the team that has the existing task.
```
Comment (Devanshshah1309) [inline docs/DeveloperGuide.md:424]: Ditto on changing all `Preconditions` to `Precondition`

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -304,19 +325,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 2a. The list is empty.
+
+  Use case ends.
+
+### Non-Functional Requirements
+
+1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+2. Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4. Does not require internet connection.
```
Comment (Devanshshah1309) [inline docs/DeveloperGuide.md:513]: Do we require it for our `help` functionality?

**Review by Devanshshah1309 — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (ling-ui) [reply docs/DeveloperGuide.md:513]: Possible, I will remove this NFR for now

**Review by Devanshshah1309 — APPROVED** [review]
LGTM

**Pull request merged**